paywall/src/unlock.js/web3Proxy.ts

```diff
@@ -1,13 +1,17 @@
-import { Web3Window, web3MethodCall } from '../windowTypes'
-import { MapHandlers, MessageHandlerTemplates } from './setupIframeMailbox'
+import { Web3Window, web3MethodCall, IframeType } from '../windowTypes'
+import {
+  MapHandlers,
+  MessageHandlerTemplates,
+  PostMessageToIframe,
+} from './setupIframeMailbox'
 import { PostMessages, MessageTypes } from '../messageTypes'
 
 let hasWeb3 = true
 
 export function enable(window: Web3Window) {
   return new window.Promise((resolve, reject) => {
     if (!window.web3 || !window.web3.currentProvider) {
-      return reject(new ReferenceError('no web3 wallet exists'))
+      return resolve('no web3 wallet')
```

Comment:
Not a blocker, but we should probably constantize and utilize that on line 77 of this file in the try

Reply:
definitely, I'll make it so